building/components/homeworld-admin-tools/src/access.py

```diff
@@ -241,11 +242,50 @@ def ssh_foreach(ops: setup.Operations, node_kind: str, *params: str):
             ops.ssh("run command on @HOST", node, *params)
 
 
+def compute_fingerprint(key: str) -> str:
+    text = subprocess.check_output(["ssh-keygen", "-l", "-f", "-"], input=key.encode(), stderr=subprocess.STDOUT).decode()
+    assert text.endswith("\n") and text.count("\n") == 1
+    return text.rstrip("\n")
+
+
+def hostkeys_by_fingerprint(node: configuration.Node, fingerprints: list):
+    keys = []
+    for line in subprocess.check_output(["ssh-keyscan", "-T", "1", "--", str(node.ip)]).decode().split("\n"):
+        if not line or line.startswith("#"): continue
+        assert line.count(" ") >= 1
+        server, key = line.split(" ", 1)
+        assert server == str(node.ip)
+        fingerprint = compute_fingerprint(key)
```

Comment:
Is there a reason to remove the comment (here, server name) so that it doesn't appear in the fingerprint, but the resulting non-comment comment (manifesting as the text `no comment`) is left in the fingerprint itself?  I think if we don't consider the comment part of the fingerprint then we should just strip it out of the fingerprint after we pass it through `ssh-keygen -lf`.  Or if it is part of the fingerprint then we should leave it intact and pass the real one to `ssh-keygen -lf`.